templates/keycloak/files/.platform.app.yaml

```diff
@@ -0,0 +1,83 @@
+name: 'app'
+
+type: 'java:11'
+
+variables:
+    env:
+        KC_VERSION: 8.0.2
+        JBOSS_HOME: /app
+        JDBC_MARIADB_VERSION: 2.2.3
+        LAUNCH_JBOSS_IN_BACKGROUND: 1
+        #DB_VENDOR: mariadb
+hooks:
+    build: |
+        set -e
+
+        if [ ! -d "$PLATFORM_CACHE_DIR"/keycloak-$KC_VERSION ]; then
+            echo "Downloading KeyCloak version ${KC_VERSION}..."
+            (
+                cd "$PLATFORM_CACHE_DIR"
+                curl -L -o $KC_VERSION.tar.gz https://downloads.jboss.org/keycloak/$KC_VERSION/keycloak-$KC_VERSION.tar.gz
+                tar -zxf $KC_VERSION.tar.gz
+                rm -f $KC_VERSION.tar.gz
+            )
+        else
+            echo "Using cached archive of Keycloak: ${KC_VERSION}..."
+        fi
+
+        echo "Installing archive of Keycloak: ${KC_VERSION}..."
+        cp -R "$PLATFORM_CACHE_DIR"/keycloak-"$KC_VERSION"/* .
+        mv standalone/configuration standalone/cfg.bkp
+
+        curl -L https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/$JDBC_MARIADB_VERSION/mariadb-java-client-$JDBC_MARIADB_VERSION.jar > modules/system/layers/base/org/mariadb/jdbc/main/mariadb-jdbc.jar
+        
+        rm -rf standalone/tmp/auth
+        rm -rf domain/tmp/auth
+    
+    deploy: |
+        if [ -f tmp/INSTALLED ]; then
+            cp standalone/cfg.bkp/* standalone/configuration/
+            cp -f tools/standalone.xml standalone/configuration/standalone.xml
+            cp -f tools/logging.properties standalone/configuration/logging.properties
+            bin/add-user-keycloak.sh -r master --user "admin" --password "password"
+            touch tmp/INSTALLED
```

Comment:
Is capitalized INSTALLED typical for other Java applications?  We're not capitalizing the touch file for any other templates.

We really should standardize how these touch files are named; we're using them on a number of templates now so consistency would be good.

Reply:
That is terrible, I agree, that is to this application and its design.

Reply:
Oh, this is building off of an INSTALLED file that the app itself already uses?  If so, that's worth a comment.